### PR TITLE
[KYUUBI-SHADED #60] Step 2/2: TSaslServerTransport use SaslServerFactory to create SaslServer

### LINKED ISSUE #60: Thrift TSaslServerTransport supports pluggable SaslServer creation method
Define a `SaslServerFactory` and allow [TSaslServerTransport](https://github.com/apache/thrift/blob/0.16.0/lib/java/src/org/apache/thrift/transport/TSaslServerTransport.java) to use `SaslServerFactory` to create `SaslServer` instead of simply calling `Sasl.createSaslServer`

```
public interface SaslServerFactory {
  SaslServer create(TSaslServerDefinition d) throws SaslException;
}
```

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

Define a `SaslServerFactory` and allow [TSaslServerTransport](https://github.com/apache/thrift/blob/0.16.0/lib/java/src/org/apache/thrift/transport/TSaslServerTransport.java) to use `SaslServerFactory` to create `SaslServer` instead of simply calling `Sasl.createSaslServer`

```
public interface SaslServerFactory {
  SaslServer create(TSaslServerDefinition d) throws SaslException;
}
```

Close #60

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-relocated-thrift/src/main/java/org/apache/thrift/transport/TSaslServerTransport.java [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+     * Get a new <code>TSaslServerTransport</code> instance, or reuse the existing one if a <code>
+     * TSaslServerTransport</code> has already been created before using the given <code>TTransport
+     * </code> as an underlying transport. This ensures that a given underlying transport instance
+     * receives the same <code>TSaslServerTransport</code>.
+     */
+    @Override
+    public TTransport getTransport(TTransport base) throws TTransportException {
+      WeakReference<TSaslServerTransport> ret = transportMap.get(base);
+      if (ret == null || ret.get() == null) {
+        LOGGER.debug("transport map does not contain key {}", base);
```
Comment (pan3793) [inline kyuubi-relocated-thrift/src/main/java/org/apache/thrift/transport/TSaslServerTransport.java:221]: this is a small fix by adding the missing placeholder `{}`
